### DISCUSSION TITLE
How to set a traffic limit for each user?

### DISCUSSION THREAD

**Discussion by skovalevskii** [XTLS/Xray-core#2727] [Q&A] [answered]
Is there a way to limit the internet traffic for each user of the Xray-core VPN installed on my server? As I got it  there is no built-in functionality to do this in the default setup. Are there any recommended plugins or alternative methods that can achieve this?

**Comment by ashad765** [comment, marked as answer]
Some web panels support traffic limit for each user.
[X-UI](https://github.com/FranzKafkaYu/x-ui)
[3X-UI](https://github.com/MHSanaei/3x-ui)
[Hiddify](https://github.com/hiddify/hiddify-config)
[Marzban](https://github.com/Gozargah/Marzban)

**Comment by skovalevskii** [comment]
I got it. Thanks a lot, I appreciate it
